File: python/model/Alarm.py
import pandas as pd


class Alarm(object):
    """
    警告类
    """

    # 标记未知
    UNKNOWN = -1
    # 误报
    FP = 0
    # 正报
    TP = 1

    def __init__(self):
        # 警告属性
        self.category = ""
        self.type = ""
        self.priority = -1
        self.rank = -1
        # 路径
        self.path = ""
        # 类名
        self.class_name = ""
        # 方法名
        self.method = ""
        # 方法签名
        self.signature = ""
        # 漏洞行位置
        self.location = -1
        # 警告标记
        self.label = Alarm.UNKNOWN
        # 所在版本
        self.version = ""

    def __eq__(self, other):
        if type(other) == type(self):
            for k in self.__dict__.keys():
                if self.__dict__[k] != other.__dict__[k]:
                    return False
            return True
        else:
            return False

    def from_dict(self, d: dict):
        for k, v in d.items():
            if k in self.__dict__.keys():
                setattr(self, k, v)
        return self

    @staticmethod
    def from_dataframe(df: pd.DataFrame) -> list:
        return [Alarm().from_dict(row) for row in df.to_dict("records")]

File: java/src/main/java/model/Alarm.java
package model;

import com.alibaba.fastjson.annotation.JSONField;
import lombok.Data;

@Data
public class Alarm {

    public static final int UNKNOWN = -1;
    public static final int FP = 0;
    public static final int TP = 1;

    private String category = "";
    private String type = "";
    private Integer priority = -1;
    private Integer rank = -1;
    private String path = "";
    @JSONField(name = "class_name")
    private String className = "";
    private String method = "";
    private String signature = "";
    private Integer location = -1;
    private Integer label = Alarm.UNKNOWN;
    private String version;
}
